### PR TITLE
log requested help URLs

### PR DESCRIPTION
**Pull request by alexbruy**
## Description
Log requested help URLs to help with debugging various issues related to new help system. Also add default help location (online User Guide for latest release) to global settings.

## Checklist

> Reviewing is a process done by project maintainers, mostly on a volunteer basis. We try to keep the overhead as small as possible and appreciate if you help us to do so by completing the following items. Feel free to ask in a comment if you have troubles with any of them.

- [x] Commit messages are descriptive and explain the rationale for changes
- [ ] Commits which fix bugs include `fixes #11111` in the commit message next to the description
- [ ] Commits which add new features are tagged with `[FEATURE]` in the commit message
- [ ] Commits which change the UI or existing user workflows are tagged with `[needs-docs]` in the commit message and containt sufficient information in the commit message to be documented
- [x] I have read the [QGIS Coding Standards](https://docs.qgis.org/testing/en/docs/developers_guide/codingstandards.html) and this PR complies with them
- [x] This PR passes all existing unit tests (test results will be reported by travis-ci after opening this PR)
- [ ] New unit tests have been added for core changes
- [x] I have run [the `scripts/prepare-commit.sh` script](https://github.com/qgis/QGIS/blob/master/.github/CONTRIBUTE.md#contributing-to-qgis) before each commit

### CODE REVIEW

**Comment by rduivenvoorde** [comment]
Nice, thanks @alexbruy 
On second thought I think it would also be usefull to show the 'key' in the msg?

Would it be an idea to make this Class also usable for Plugin writers? 
Like they add their own key-value settings and then call this class (BUT then using another base url)?
Or is it better to let them use  plain QDesktopServices::openUrl(...)

File: resources/qgis_global_settings.ini [hunk 1]
```diff
@@ -28,3 +28,8 @@ connections-xyz\OpenStreetMap\url=http://a.tile.openstreetmap.org/{z}/{x}/{y}.pn
 connections-xyz\OpenStreetMap\username=
 connections-xyz\OpenStreetMap\zmax=19
 connections-xyz\OpenStreetMap\zmin=0
+
+[help]
+# Default help location to inlude
```
Comment (nyalldawson) [inline resources/qgis_global_settings.ini]: "include"

File: resources/qgis_global_settings.ini [hunk 2]
```diff
@@ -28,3 +28,8 @@ connections-xyz\OpenStreetMap\url=http://a.tile.openstreetmap.org/{z}/{x}/{y}.pn
...
+# for now this is online version fo the User Guide for latest (LTR) release
```
Comment (nyalldawson) [inline resources/qgis_global_settings.ini]: "of"

**Comment by nyalldawson** [comment]
I'm starting to wonder if we need some way for code to "register" settings keys, alongside some descriptive text explaining the setting. This could be used to auto populate the whole default settings ini with both default values AND descriptive text, and could also be used as a tooltip in the advanced settings editor in options.

@elpaso @nathanw2 @3nids what do you think?

**Comment by alexbruy** [comment]
@nyalldawson thanks, fixed.

**Comment by alexbruy** [comment]
@rduivenvoorde added, please check if this is what you need.

**Comment by rduivenvoorde** [comment]
@alexbruy Thanks! All fine. Only think I see, if you start a new config and did not 'save' default settings (https://issues.qgis.org/issues/17670) you still get Oopsies, without seeing debug message...
In that case line 45: still has an empty value:
`QStringList paths = settings.value( QStringLiteral( "help/helpSearchPath" ) ).toStringList();`

What about also showing msg in  this line(s)? Would mean that it is clear if 
`https://docs.qgis.org/$qgis_short_version/$qgis_locale/docs/user_manual/`
is available AND to what is resolved too.
That would mean though that every call to help would show the url.. is maybe too much.
BUT adding a message when 
`paths.isEmpty()` is actually true IS good too see in my opinion.
@alexbruy agree?

**Comment by rduivenvoorde** [comment]
@alexbruy yep, sorry for my last messy comment. 

My last ;-) proposal is indeed to write in the log message that help location is not configured yet.

**Comment by alexbruy** [comment]
@rduivenvoorde added.

**Comment by NathanW2** [comment]
@elpaso I considered using sqlite as a settings store with a custom table for QGIS 3 kind of annoyed at myself for not looking into it more now.  That would have been really handy to have.

**Comment by rduivenvoorde** [comment]
@alexbruy Good to go for me. Please pull.

**Pull request merged**